crates/ethereum/src/lib.rs

```diff
@@ -1,111 +1,354 @@
-use std::convert::TryFrom;
+use pathfinder_common::{BlockHash, BlockNumber, EthereumChain, StateCommitment};
+use primitive_types::{H160, H256, U256};
+use stark_hash::Felt;
 
-use anyhow::{Context, Result};
+pub mod core_addr {
+    use const_decoder::Decoder;
 
-use pathfinder_common::{
-    EthereumBlockHash, EthereumBlockNumber, EthereumLogIndex, EthereumTransactionHash,
-    EthereumTransactionIndex,
-};
-
-pub mod contract;
-pub mod log;
-pub mod provider;
-pub mod state_update;
+    pub const MAINNET: [u8; 20] = Decoder::Hex.decode(b"c662c410C0ECf747543f5bA90660f6ABeBD9C8c4");
+    pub const TESTNET: [u8; 20] = Decoder::Hex.decode(b"de29d060D45901Fb19ED6C6e959EB22d8626708e");
+    pub const TESTNET2: [u8; 20] = Decoder::Hex.decode(b"a4eD3aD27c294565cB0DCc993BDdCC75432D498c");
+    pub const INTEGRATION: [u8; 20] =
+        Decoder::Hex.decode(b"d5c325D183C592C94998000C5e0EED9e6655c020");
+}
 
-#[derive(Debug, Clone, PartialEq, Hash, Eq)]
-pub struct BlockOrigin {
-    pub hash: EthereumBlockHash,
-    pub number: EthereumBlockNumber,
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EthereumStateUpdate {
+    pub global_root: StateCommitment,
```

Comment:
Maybe `state_commitment` instead of `global_root` here? (https://docs.starknet.io/documentation/architecture_and_concepts/State/starknet-state/#state_commitment)

Reply:
The [contract's](https://etherscan.io/address/0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4#readProxyContract) method is `stateRoot()`, I suggest we keep naming consistent. It is clear though what `StateCommitment` stands for.

Reply:
The reason the L1 calls it `stateRoot` is because its what they called it originally. I imagine if they could redo it now, they would call it `stateCommitment`